src/ui_ng/src/app/project/chart-detail/chart-detail.component.ts

```diff
@@ -0,0 +1,55 @@
+import { RoleMapping } from './../../shared/shared.const';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Component, OnInit } from "@angular/core";
+import { Project } from '../project';
+import { SessionService } from './../../shared/session.service';
+import { SessionUser } from './../../shared/session-user';
+
+@Component({
+  selector: "project-chart-detail",
+  templateUrl: "./chart-detail.component.html",
+  styleUrls: ["./chart-detail.component.scss"]
+})
+export class ChartDetailComponent implements OnInit {
+
+  projectId: number | string;
+  project: Project;
+  chartName: string;
+  chartVersion: string;
+  currentUser: SessionUser;
+  hasProjectAdminRole: boolean;
+  roleName: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private session: SessionService
+  ) {}
+
+  ngOnInit() {
+    // Get projectId from route params snapshot.
+    this.projectId = +this.route.snapshot.params['id'];
```

Comment:
should define a variable.  this.route.snapshot.

Reply:
I think using `this.route.snapshot` is a convention, rename it may confuse people.